lib/files.js

```diff
@@ -10,14 +10,24 @@ var path = require('path');
 var options = require('./args.js');
 var log = require('davlog');
 var verify = require('./verify.js');
+var hooks = require('./hooks');
 
 var putBall = function(info, callback) {
     var localBall = path.join(options.dir, info.path);
-    process.nextTick(function() {
-        mkdirp(path.dirname(localBall), function() {
-            process.nextTick(function() {
-                info.tarball = localBall;
-                verify.verify(info, callback);
+    info.tarball = localBall;
+
+    hooks.tarball(info, callback, function () {
```

Comment:
From the docs, the callback here should have an `error` and a `shouldSave` right? Then bail if `shouldSave` is false?